src/sqlfluff/rules/L036.py

```diff
@@ -306,25 +306,27 @@ def _fixes_for_move_after_select_clause(
                         )
                         fixes += [LintFix.delete(seg) for seg in to_delete]
 
-                        # The select_clause is immediately followed by a
-                        # newline. Delete the newline in order to avoid leaving
-                        # behind an empty line after fix, *unless* we stopped
-                        # due to something other than a newline.
-                        delete_last_newline = select_children[
-                            start_idx - len(to_delete) - 1
-                        ].is_type("newline")
-
-                        # Delete the newline if we decided to.
-                        if delete_last_newline:
-                            fixes.append(
-                                LintFix.delete(
-                                    select_stmt.segments[after_select_clause_idx],
+                        if to_delete:
```

Comment:
Review this with whitespace hiding and you'll see change is very simple:

<img width="370" alt="image" src="https://user-images.githubusercontent.com/10931297/157333017-eedfeb86-c437-4d74-a255-52d00af74728.png">